File: two-sum.js
var twoSum = function (nums, target) {
  const numsMap = new Map();

  for (let i = 0; i < nums.length; i++) {
    const remIndex = numsMap.get(target - nums[i]);
    if (remIndex !== undefined && remIndex !== i) {
      return [remIndex, i];
    }
    numsMap.set(nums[i], i);
  }

  return [];
};

console.log(twoSum([2,7,11,15], 9))

File: python/two-sum.py


def twoSum(nums, target):
    hashmap = {}

    for i, num in enumerate(nums):
        reminder = target -  num
        if reminder in hashmap and hashmap[reminder] != i:
            return [i, hashmap[reminder]]
        hashmap[num] = i


print(twoSum([2,7,11,15], 9))